core/app/models/spree/return_authorization.rb

```diff
@@ -61,6 +62,10 @@ def can_cancel_return_items?
 
     private
 
+      def authorize_return
+        order.authorize_return! unless order.nil?
```

Comment:
I never saw this specific collective/interactor gem before myself (looks cool!), but I personally adopted much of this style of writing Rails app (I write my own 'context' objects which are basically nearly the same pattern as what the 'interactor' pattern does above) after having read *Lean Architecture: for Agile Software Development* (Coplien,  Bjørnvig 2011). Coplien is prolific on the subject of *domain context interaction* (or *data, context interaction* depending on whom you ask). 

In my view moving towards this style is an industry-wide thing that all the top Rails devs I know are moving towards (except one single significant voice of opposition who happens to be the creator of Rails himself). DHH himself has been less vociferous in his opposition to patterns like these the last few years, falling back on arguments like "well if it works for you then do it, but Rails shouldn't dictate it"

My feeling is that a large codebase like Spree really *must* move in this direction in order to survive long-term, since no large app that has poor domain modeling is ever fun or cheap to work with.